src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -178,23 +187,12 @@ struct patcher
     bd_data_ptr[2] = (bd_data_ptr[2] & 0xFFFF0000) | (base_address >> 32);            // NOLINT
   }
 
-  // TODO reuse patch_ctrl48 for both DPU sequence and transaction buffer
-  // Need change assembler first
-  void patch_tansaction_ctrlpkt_48(uint32_t* bd_data_ptr, uint64_t patch)
-  {
-    uint64_t val = *reinterpret_cast<uint64_t*>(bd_data_ptr);
-    val &= 0x0000FFFFFFFFFFFF;                                                        // NOLINT
-    val += patch;
-    *bd_data_ptr = static_cast<uint32_t>(val);
-
-    auto higher16BitVal = static_cast<uint16_t>((val & 0x0000FFFF00000000) >> 32);    // NOLINT
-    auto pHigher16Bit = reinterpret_cast<uint16_t*>(bd_data_ptr + 1);
-    *pHigher16Bit = higher16BitVal;
-  }
-
   void
-  patch(uint8_t* base, uint64_t patch)
+  patch(uint8_t* base, uint64_t patch, buf_type type)
   {
+    if (type != m_buf_type)
```

Comment:
With this line of change, now all 4 test cases from xcompiler-team are working.
Without this change, case-conv1 does not work, This is documented in https://jira.xilinx.com/browse/VITIS-11623